src/script/page/RightSidebar/MessageDetails/MessageDetails.tsx

```diff
@@ -0,0 +1,276 @@
+
+  const getLikes = useCallback(async (reactions: UserReactionMap) => {
+    const currentLikes = Object.keys(reactions);
+    const usersLikes = await userRepository.getUsersById(currentLikes.map(likeId => ({domain: '', id: likeId})));
+    const sortedUsersLikes = usersLikes.sort(sortUsers);
+
+    setLikeUsers(sortedUsersLikes);
+  }, []);
+
+  const receipts = messageEntity?.readReceipts() || [];
+
+  const receiptTimes = useMemo(() => {
+    const userIds: QualifiedId[] = receipts.map(({userId, domain}) => ({domain: domain || '', id: userId}));
+    userRepository.getUsersById(userIds).then((users: User[]) => {
+      setReceiptUsers(users.sort(sortUsers));
+    });
+
+    return receipts.reduce<Record<string, string>>((times, {userId, time}) => {
+      times[userId] = formatTime(time);
+      return times;
+    }, {});
+  }, []);
+
+  const sentFooter = timestamp ? formatTime(timestamp) : '';
+
+  const receiptsTitle = t(
+    'messageDetailsTitleReceipts',
+    messageEntity?.expectsReadConfirmation ? formatUserCount(receiptUsers) : '',
+  );
+  const likesTitle = t('messageDetailsTitleLikes', formatUserCount(likeUsers));
+
+  const panelTitle = useMemo(() => {
+    if (!supportsReceipts) {
+      return likesTitle;
+    }
+
+    if (!supportsLikes) {
+      return receiptsTitle;
+    }
+
+    return t('messageDetailsTitle');
+  }, [supportsReceipts, supportsLikes]);
+
+  const showTabs = supportsReceipts && supportsLikes;
+
+  const messageTimestamp = isContentMessage(messageEntity) && messageEntity.edited_timestamp();
+  const editedFooter = messageTimestamp ? formatTime(messageTimestamp) : '';
+
+  const onReceipts = () => setIsReceiptsOpen(true);
+
+  const onLikes = () => setIsReceiptsOpen(false);
+
+  useEffect(() => {
+    if (supportsLikes && isContentMessage(messageEntity)) {
+      const messageReactions = messageEntity.reactions();
+      getLikes(messageReactions);
+    }
+  }, [getLikes, supportsLikes, messageEntity]);
+
+  useEffect(() => {
+    amplify.subscribe(WebAppEvents.CONVERSATION.MESSAGE.UPDATED, (oldId: string, updatedMessageEntity: Message) => {
+      // listen for any changes to local message entities.
+      // if the id of the message being viewed has changed, we store the new ID.
+      if (oldId === messageId) {
+        updateEntity(updatedMessageEntity);
+        setMessageId(updatedMessageEntity.id);
+
+        if (supportsLikes && isContentMessage(updatedMessageEntity)) {
+          const messageReactions = updatedMessageEntity.reactions();
+          getLikes(messageReactions);
+        }
+      }
+    });
+  }, [messageId, supportsLikes]);
+
+  return (
+    <div id="message-details" className="panel__page message-details panel__page--visible">
+      <PanelHeader
+        onClose={onClose}
+        title={panelTitle}
+        showBackArrow={false}
+        titleDataUieName="message-details-title"
+        closeBtnTitle={t('accessibility.rightPanel.close')}
```

Comment:
Add goBackTitle and a more meaningful  closeBtnTitle, the default value of closeBtnTitle is already set to t('accessibility.rightPanel.close') inside PanelHeader so unless Astrid wants to specify something specific here you can skip this prop.